src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java

```diff
@@ -224,7 +224,10 @@ public void connect() {
             } else {
                 jmsCtxt = mqConnFactory.createContext(JMSContext.SESSION_TRANSACTED);
             }
-
+            
+            if (jmsCons != null) {
+                jmsCons.close();
```

Comment:
The principle here (of checking if there is something already present and cleaning it up if so) is correct, but I think this is the wrong place to do it. 

You're checking inside `JMSReader`.

The new config is provided in `MQSourceTask` here: 
https://github.com/ibm-messaging/kafka-connect-mq-source/blob/0119b06302d897609e925df0df091845330159a1/src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceTask.java#L61

which responds by creating a new `JMSReader`. 
https://github.com/ibm-messaging/kafka-connect-mq-source/blob/0119b06302d897609e925df0df091845330159a1/src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceTask.java#L80

Because of this, your not-null check will never pass, and your `jmsCons.close()` line will never be executed. 

Because a new `JMSReader` is always created, that new JMSReader instance will always have a null jmsCons instance inside it.